include/cetl/pf17/any.hpp

```diff
@@ -0,0 +1,46 @@
+/// @file
+/// Defines the C++17 `std::any` type and several related entities.
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PF17_ANY_HPP_INCLUDED
+#define CETL_PF17_ANY_HPP_INCLUDED
+
+#include "attribute.hpp"
+
+#include <typeinfo>   // We need this for `std::bad_cast`.
+#include <exception>  // We need this even if exceptions are disabled for `std::terminate`.
+
+namespace cetl  // NOLINT(*-concat-nested-namespaces)
+{
+namespace pf17
+{
+
+#if defined(__cpp_exceptions) || defined(CETL_DOXYGEN)
+
+/// A polyfill for `std::bad_any_cast`.
+/// This is only available if exceptions are enabled (`__cpp_exceptions` is defined).
+class bad_any_cast : public std::bad_cast
```

Comment:
It makes sense that this class is here, as it polyfills `std::bad_any_cast`, but I wonder if it's going to be a problem later on when we need to add a specialization of our `cetl::any<>` into this file, because that would require this header to depend on `cetl/any.hpp`, which creates a recursive dependency.

If this is a valid concern, we might be better off moving everything that is needed for `cetl/any.hpp` into its own header.

Reply:
Yes, I though about it, but decided to tackle this when I start working on the specialization. Is it fine to postpone?

Reply:
okay